includes/Admin/Product_Categories.php

```diff
@@ -143,10 +148,30 @@ public function get_google_product_category_field_title() {
 	 *
 	 * @since 2.1.0-dev.1
 	 *
-	 * @return string
+	 * @param int $term_id term ID
+	 * @param int $tt_id term taxonomy ID
+	 * @param string $taxonomy Taxonomy slug
 	 */
-	public function save_google_product_category() {
+	public function save_google_product_category( $term_id, $tt_id, $taxonomy ) {
+
+		$google_product_category_id = Framework\SV_WC_Helper::get_posted_value( self::FIELD_GOOGLE_PRODUCT_CATEGORY_ID );
+
+		\SkyVerge\WooCommerce\Facebook\Product_Categories::update_google_product_category_id( $term_id, $google_product_category_id );
+
+		$term = get_term( $term_id, $taxonomy, ARRAY_A );
+
+		// get the products in the category being saved
+		$product_ids = wc_get_products(
+			[
+				'return'   => 'ids',
+				'category' => [ $term['slug'] ],
+			]
+		);
+
+		if ( ! empty( $product_ids ) ) {
 
+			facebook_for_woocommerce()->get_products_sync_handler()->create_or_update_products( $product_ids );
```

Comment:
We should only pass the ID of simple products and product variations to `create_or_update_products()`. That method doesn't check the type of the product and we may end up using the Batch API to try to sync info for a variable product instead of the individual variations.

Simon and I ran into that accidentally on a support conversation and the plugin started creating a variation with zero price in addition to the normal variations of variable products.

Reply:
Updated in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1551/commits/7ee343bad8659462d54dcb4f1096cca6e68ac08a.